src/network/networkd-link.c

```diff
@@ -902,7 +906,7 @@ static int link_request_set_neighbors(Link *link) {
         assert(link->network);
         assert(link->state != _LINK_STATE_INVALID);
 
-        link_set_state(link, LINK_STATE_CONFIGURING);
```

Comment:
You should be able to leave this once we correct the invariants on link_check_ready(). Just in case we decide to re-arrange the ordering of the link_set_* functions in the future. There is no reason why link_set_neighbors should depend on link_set_addresses setting this state.